### PR TITLE
Add support for memory utilization in HorizontalPodAutoscaler

### LINKED ISSUE #529: [Enhancement][opensearch-dashboards] Add support for memory utilization in HorizontalPodAutoscaler
**Is your feature request related to a problem? Please describe.**
Our Dashboards are fairly memory intensive, so we have had to modify the HPA created by the helm chart to support autoscaling based on memory utilization. 

**Describe the solution you'd like**
This feature request would enable support for memory utilization as for HPA metrics.

### PR DESCRIPTION
**Pull request by Simple-Analysis**
### Description
Add support for memory utilization in HorizontalPodAutoscaler.
 
### Issues Resolved
#529 
 
### Check List
- [x] Commits are signed per the DCO using --signoff

For any changes to files within Helm chart directories:
- [x] Helm chart version bumped
- [x] Helm chart `CHANGELOG.md` updated to reflect change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/helm-charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: charts/opensearch-dashboards/CHANGELOG.md [hunk 1]
```diff
@@ -13,9 +13,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 ### Security
 ---
+## [2.17.0]
```
Comment (prudhvigodithi) [inline charts/opensearch-dashboards/CHANGELOG.md:16]: Hey @Simple-Analysis can you please change this `HorizontalPodAutoscaler` to 2.16.1?

Comment (TheAlgo) [reply charts/opensearch-dashboards/CHANGELOG.md:16]: +1 This will break upgrades right?

File: charts/opensearch-dashboards/values.yaml [hunk 2]
```diff
@@ -199,7 +199,9 @@ autoscaling:
   enabled: false
   minReplicas: 1
   maxReplicas: 10
-  targetCPUUtilizationPercentage: 80
+  # The legacy `targetCPUUtilizationPercentage` key has been deprecated in favor of `targetCPU`
+  targetCPU: "80"
```
Comment (TheAlgo) [inline charts/opensearch-dashboards/values.yaml:203]: @Simple-Analysis can we also add a 1 liner description the [README](https://github.com/opensearch-project/helm-charts/blob/main/charts/opensearch/README.md) which will help users better use this feature?

File: charts/opensearch-dashboards/CHANGELOG.md (hunk 1 above)
Comment (Simple-Analysis) [reply charts/opensearch-dashboards/CHANGELOG.md:16]: Sure thing!

```suggestion
## [2.16.1]
```

**Review by TheAlgo — CHANGES_REQUESTED** [review]

File: charts/opensearch-dashboards/values.yaml (hunk 2 above)
Comment (Simple-Analysis) [reply charts/opensearch-dashboards/values.yaml:203]: > @Simple-Analysis can we also add a 1 liner description the [README](https://github.com/opensearch-project/helm-charts/blob/main/charts/opensearch/README.md) which will help users better use this feature?

Definitely. I added descriptions for the autoscaling parameters.

File: charts/opensearch-dashboards/Chart.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 2.16.0
+version: 2.17.0
```
Comment (Simple-Analysis) [inline charts/opensearch-dashboards/Chart.yaml:18]:
```suggestion
version: 2.16.1
```
Update version per @prudhvigodithi's [requested change](https://github.com/opensearch-project/helm-charts/pull/530#discussion_r1546765134)

**Review by TheAlgo — APPROVED** [review]
Looks good to me now!

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Thanks @TheAlgo.

**Pull request merged**